### PR TITLE
`view_as_windows` incorrectly assumes that a contiguous array is needed 

### LINKED ISSUE #3088: `view_as_windows` incorrectly assumes that a contiguous array is needed
See https://github.com/numpy/numpy/pull/10771#issuecomment-390881027

Mea culpa.

### PR DESCRIPTION
**Pull request by sciunto**
## Description

Fix #3088 

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/util/tests/test_shape.py [hunk 1]
```diff
@@ -143,14 +143,6 @@ def test_view_as_windows_with_skip():
     assert_equal(C.shape, (1, 1, 2, 2))
 
 
-def test_views_non_contiguous():
-    A = np.arange(16).reshape((4, 4))
-    A = A[::2, :]
-
-    assert_warns(RuntimeWarning, view_as_blocks, A, (2, 2))
-    assert_warns(RuntimeWarning, view_as_windows, A, (2, 2))
```
Comment (jni) [inline skimage/util/tests/test_shape.py]: @sciunto rather than removing this test, can you make sure it works correctly? Other than that, LGTM!

Comment (sciunto) [reply skimage/util/tests/test_shape.py]: you mean by adding a test? (then how?) or just checking manually?

Comment (jni) [reply skimage/util/tests/test_shape.py]: No no, I just meant check that it runs without error. Although checking manually is not too hard:

```python
image = np.arange(8)
image_non_contiguous = image[::2]
expected = np.array([[0, 2], [2, 4], [4, 6]])
assert_equal(view_as_windows(image_non_contiguous, (2,)), expected)
```

For a clean 2D test, just spot-check one of the values I guess:

```python
image = np.zeros((6, 6))
image[::2, ::2] = np.arange(9).reshape((3, 3))
assert_equal(view_as_windows(image[::2, ::2], (2, 2))[1, 1], [[4, 5], [7, 8]])
```

**Comment by sciunto** [comment]
I reverted the commit and adapted the test.

**Comment by emmanuelle** [comment]
Thank you @sciunto ! Merging.

**Comment by sciunto** [comment]
i'm glad to see you around @emmanuelle :)

**Pull request merged**